nucypher/cli/characters/stake.py

```diff
@@ -101,133 +190,84 @@ def stake():
 
 
 @stake.command()
-@_api_options
-@nucypher_click_config
-def accounts(click_config,
-
-             # API Options
-             poa, registry_filepath, config_file, provider_uri, staking_address):
+@group_character1_options
+@option_config_file
+@group_general_config
+def accounts(general_config, character1_options, config_file):
     """
     Show ETH and NU balances for stakeholder's accounts.
     """
-
-    ### Setup ###
-    emitter = _setup_emitter(click_config)
-
-    STAKEHOLDER, blockchain = _create_stakeholder(config_file, provider_uri, poa, registry_filepath, staking_address,
-                                                  beneficiary_address=None, allocation_filepath=None)
-    #############
-
+    emitter = _setup_emitter(general_config)
+    STAKEHOLDER = character1_options.create_object(emitter, config_file)
     painting.paint_accounts(emitter=emitter, balances=STAKEHOLDER.wallet.balances)
 
 
 @stake.command('set-worker')
-@_worker_options
-@click.option('--force', help="Don't ask for confirmation", is_flag=True)
-@nucypher_click_config
-def set_worker(click_config,
-
-               # Worker Options
-               poa, registry_filepath, config_file, provider_uri, staking_address, hw_wallet,
-               beneficiary_address, allocation_filepath,
-               worker_address, 
-               
-               # Other options
-               force):
+@group_character2_options
+@option_config_file
+@option_force
+@group_general_config
+@option_worker_address
+def set_worker(general_config, character2_options, config_file, force, worker_address):
     """
     Bond a worker to a staker.
     """
-
-    ### Setup ###
-    emitter = _setup_emitter(click_config)
-
-    STAKEHOLDER, blockchain = _create_stakeholder(config_file, provider_uri, poa, registry_filepath, staking_address,
-                                                  beneficiary_address=beneficiary_address,
-                                                  allocation_filepath=allocation_filepath)
-    #############
-
+    emitter = _setup_emitter(general_config)
+    STAKEHOLDER = character2_options.create_object(emitter, config_file)
+    blockchain = character2_options.get_blockchain()
     economics = STAKEHOLDER.economics
 
-    client_account, staking_address = handle_client_account_for_staking(emitter=emitter,
-                                                                        stakeholder=STAKEHOLDER,
-                                                                        staking_address=staking_address,
-                                                                        individual_allocation=STAKEHOLDER.individual_allocation,
-                                                                        force=force)
+    client_account, staking_address = handle_client_account_for_staking(
+        emitter=emitter,
+        stakeholder=STAKEHOLDER,
+        staking_address=character2_options.character1_options.staking_address,
```

Comment:
I don't understand what's going on here? What do `Character1` and `Character2` mean?

Reply:
As I mentioned in the general description, the names are subject to change. Basically, the data flow in `stake.py` implies that there are two option groups used to create a character, one (the smaller one) used in `list` in `accounts`, and the other (the bigger one) in the rest of the commands. I suspect there may be some actual terms that can be used to denote them.

Reply:
...So this is an "option group"?

Reply:
Yep. The option group thing is essentially unchanged from the Derek's PR, but they're passed as a single object to the command functions instead of all the constituents separately. The common pattern seems to be [[options used to create/find a config] options used to create the character], but in `stake.py` there are two scenarios of character creation.

Reply:
Do the names "Configuration Options" and "Character Options" fit?